plugins/woocommerce/src/Internal/Admin/Settings/PaymentsProviders/PaymentGateway.php

```diff
@@ -239,7 +254,33 @@ public function is_enabled( WC_Payment_Gateway $payment_gateway ): bool {
 	 * @return bool True if the payment gateway needs setup, false otherwise.
 	 */
 	public function needs_setup( WC_Payment_Gateway $payment_gateway ): bool {
-		return filter_var( $payment_gateway->needs_setup(), FILTER_VALIDATE_BOOLEAN );
+		try {
```

Comment:
It is better to log when dealing with a wrong type since `wc_string_to_bool` will error out if it is not a bool or a string.